mesh/mesh_readers.cpp

```diff
@@ -1540,7 +1540,11 @@ void Mesh::ReadGmshMesh(std::istream &input, int &curved, int &read_gf)
                   // non-positive attributes are not allowed in MFEM
                   if (phys_domain <= 0)
                   {
-                     MFEM_ABORT("Non-positive element attribute in Gmsh mesh!");
+                     MFEM_ABORT("Non-positive element attribute in Gmsh mesh!\n"
+                                "Use \"Physical Curve\", \"Physical Surface\","
```

Comment:
Maybe add a sentence like this here based on your reply in https://github.com/mfem/mfem/issues/2045#issuecomment-776882499:

>  Note that by default Gmsh sets the element tags (attributes) to '0' but MFEM requires that they be positive integers.

to make it clear that Gmsh tags are equivalent to MFEM attributes and that the error is due to default Gmsh behavior or user error.

Also, you could add the lower bound for the attribute number, e.g. like this:

>...to set tags/attributes >= 1...

Reply:
Good suggestion. I've expanded it a little more.